DatadogSessionReplay/Sources/SessionReplay+Internal.swift

```diff
@@ -0,0 +1,152 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2023-Present Datadog, Inc.
+ */
+
+import Foundation
+import DatadogInternal
+import UIKit
+
+@_spi(Internal) public enum SessionReplayElement {
+    case specificElement (subtreeStrategy: SessionReplayNodeSubtreeStrategy, attributes: SessionReplayViewAttributes, builder: GenericTextWireframesBuilder)
+    case invisibleElement
+}
+
+@_spi(Internal) open class CustomNodeRecorder: NodeRecorder {
+    public let identifier = UUID()
+    
+    public init () {}
+
+    internal func semantics(of view: UIView, with attributes: ViewAttributes, in context: ViewTreeRecordingContext) -> NodeSemantics? {
+        let element = getSessionReplayElement(view: view, attributes: attributes, context: context)
+        switch element {
+        case .invisibleElement:
+            return InvisibleElement.constant
+        case .none:
+            return nil
+        case .specificElement(let subtreeStrategy, let attributes, let builder):
+            let nodes = [
+                Node(viewAttributes: attributes, wireframesBuilder: builder)
+            ]
+            return SpecificElement(subtreeStrategy: subtreeStrategy, nodes: nodes)
+        }
+    }
+
+    // will be overriden with actual implementation
+    open func getSessionReplayElement(
```

Comment:
Maybe this function can be passed in the constructor instead of being overriden, I think it'll be a bit cleaner

Reply:
Maybe it makes sense to make it a protocol with default implementation of `semantics(of` (via extension) in that case? 🤔